### ISSUE TITLE
how to try an alternative for failed Try with a new call that returns a Try

### ISSUE THREAD

**Issue by mathiasbn** [louthy/language-ext#270]
So what i'm looking for is something like what other frameworks (jvm) calls 

`public Try<T> OrElseTry(this Try<T>, Func<T> alternative)`

So if the `this Try` is in Error then alternative is called
Is that possible? It seems resonable.

**Comment by louthy** [comment]
Surprisingly no, that is not in there yet.  The other monadic types use the `||` operator to achieve this, but because `Try<A>` is a `delegate` we can't do that.  

So for example with `Option<int>` you can do this:
```c#
    Option<int> x = None;
    Option<int> y = Some(123);

    var z = x || y;   

    // z == Some(123)
```
The functionality already exists in the type-class instance for `Try<A>` called `MTry<A>`:
```c#
    default(MTry<A>).Plus(a, b);
```
You can implement your own extension trivially until I get a chance to do an update:
```c#
    using LanguageExt;
    using LanguageExt.ClassInstances;

    public static class YourTryExtensions
    {
        public static Try<A> Or<A>(this Try<A> self, Try<A> alt) =>
            default(MTry<A>).Plus(self, alt);
    }
```

**Comment by mathiasbn** [comment]
Thank you for quick reply. I can live with that extension. I feared that I missed something, I'm not used to frameworks so Haskell'ish near.

**Comment by mathiasbn** [comment]
And then a follow up.
I found that I actually could type it better with a TryOption.Or(Try)

To add a little context. 
The first call is searching for a contact. That call could fail, hence the Try part. But it could also be "not found", hence the option part.
If it is not found, I will create it, hence the Or(Try) part.

How would I update my Extension to that? 
(Sorry, I find this orthodox functional style a bit more difficult than other functional libs i've tried...)

**Comment by louthy** [comment]
On my phone, so apologies for the bad formatting/ errors:

static Try<A> Or<A>(this TryOption<A> self, Try<A> alt) =>
self.Match(
Some: x => Try(x),
None: () => alt,
Fail: ex => alt);

**Comment by louthy** [comment]
@mathiasbn Hi Mathias, a bit late, but this is fixed in `v2.2.7-beta`

There is now a `choice` function in the `Prelude` for `Try`, `TryOption`, `TryAsync`, `TryOptionAsync`, and `Task`.

For example:
```c#
var ma = Try<int>(() => throw new Exception());
var mb = Try<int>(() => throw new Exception());
var mc = Try(() => 123);

var res = choice(ma, mb, mc);

Assert.True(res.IfFail(0) == 123);
```
